### PR TITLE
Improve subkey error reporting

### PR DESCRIPTION
**Pull request by NikVolf**
```
     Running `target/debug/subkey generate --words 1234123`

Error:
        Invalid number of words given for phrase: must be 12/15/18/21/24

```

### CODE REVIEW

File: bin/utils/subkey/src/main.rs [hunk 1]
```diff
@@ -236,15 +236,37 @@ fn main() {
...
 
-fn execute<C: Crypto>(matches: ArgMatches)
+#[derive(Debug, derive_more::Display, derive_more::From)]
+enum Error {
+	Static(&'static str),
+	Io(std::io::Error),
+	Formatted(String),
+}
+
+fn execute_proc<C: Crypto>(matches: ArgMatches)
```
Comment (bkchr) [inline bin/utils/subkey/src/main.rs]: Just use `fn main() -> Result<(), Error>`

Comment (NikVolf) [reply bin/utils/subkey/src/main.rs]: It does not really pretty-print error this way, see `nv-sk-err-msin` branch

Comment (bkchr) [reply bin/utils/subkey/src/main.rs]: If you do this: https://github.com/paritytech/substrate/blob/c3a4c55cfc3a5d6f236dc184ea76f5b9ebe1eae4/client/network/src/error.rs#L51

The error message should look good.

Comment (NikVolf) [reply bin/utils/subkey/src/main.rs]: Better, but still worse than here if you ask me (updated `nv-sk-err-msin` branch)

Comment (bkchr) [reply bin/utils/subkey/src/main.rs]: The only different I see is that there is no new line after `Error:` in the output? I think hat is okay and less code is always better, IMHO^^

Comment (NikVolf) [reply bin/utils/subkey/src/main.rs]: @bkchr updated

File: bin/utils/subkey/src/main.rs [hunk 2]
```diff
@@ -270,7 +270,24 @@ fn get_uri(match_name: &str, matches: &ArgMatches) -> String {
 	}
 }
 
-fn execute<C: Crypto>(matches: ArgMatches)
+#[derive(derive_more::Display, derive_more::From)]
```
Comment (bkchr) [inline bin/utils/subkey/src/main.rs]: Could you please change the function above to return an error as well? (get_uri)

**Pull request merged**